### PR TITLE
OPENEUROPA-2106: Rework the Poetry client service.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: modules/oe_translation_poetry/src/Poetry.php [hunk 1]
```diff
@@ -90,29 +139,33 @@ public function __construct(array $settings, ConfigFactoryInterface $configFacto
       'service.password' => Settings::get('poetry.service.password'),
       'notification.username' => Settings::get('poetry.notification.username'),
       'notification.password' => Settings::get('poetry.notification.password'),
-      'logger' => $loggerChannel,
+      'notification.endpoint' => Url::fromRoute('oe_translation_poetry.notifications')->setAbsolute()->toString(),
```
Comment (dxvargas) [inline modules/oe_translation_poetry/src/Poetry.php]: Missing route definition.

File: modules/oe_translation_poetry/src/Poetry.php [hunk 2]
```diff
@@ -90,29 +139,33 @@ public function __construct(array $settings, ConfigFactoryInterface $configFacto
...
+  /**
+   * Delegates to the Poetry library all calls made to this service.
+   *
+   * {@inheritdoc}
+   */
+  public function __call($name, $arguments) {
+    $this->initialize();
+    if (method_exists($this->poetryClient, $name)) {
+      return call_user_func_array([$this->poetryClient, $name], $arguments);
+    }
```
Comment (dxvargas) [inline modules/oe_translation_poetry/src/Poetry.php]: And if method is not defined...? Shouldn't it throw an error?

File: modules/oe_translation_poetry/src/Poetry.php (hunk 1 above)
Comment (upchuk) [reply modules/oe_translation_poetry/src/Poetry.php]: True, fixed.

File: modules/oe_translation_poetry/src/Poetry.php (hunk 2 above)
Comment (upchuk) [reply modules/oe_translation_poetry/src/Poetry.php]: yes, just as if you call a missing method on any class.

**Pull request merged**